cmd/dep/status.go

```diff
@@ -304,18 +348,26 @@ type MissingStatus struct {
 	MissingPackages []string
 }
 
-func runStatusAll(ctx *dep.Ctx, out outputter, p *dep.Project, sm gps.SourceManager) (bool, bool, error) {
-	var digestMismatch, hasMissingPkgs bool
+// errorStatus contains information about error and number of status failures.
+type errorStatus struct {
+	err error
+	// count is for counting errors due to which we don't fail completely, but
+	// return partial results with missing/unknown data.
+	count int
+}
 
+func runStatusAll(ctx *dep.Ctx, out outputter, p *dep.Project, sm gps.SourceManager) (digestMismatch bool, hasMissingPkgs bool, errStatus errorStatus) {
 	if p.Lock == nil {
-		return digestMismatch, hasMissingPkgs, errors.Errorf("no Gopkg.lock found. Run `dep ensure` to generate lock file")
+		errStatus.err = errors.Errorf("no Gopkg.lock found. Run `dep ensure` to generate lock file")
+		return digestMismatch, hasMissingPkgs, errStatus
 	}
 
 	// While the network churns on ListVersions() requests, statically analyze
 	// code from the current project.
 	ptree, err := pkgtree.ListPackages(p.ResolvedAbsRoot, string(p.ImportRoot))
 	if err != nil {
-		return digestMismatch, hasMissingPkgs, errors.Wrapf(err, "analysis of local packages failed")
+		errStatus.err = errors.Wrapf(err, "analysis of local packages failed")
+		return digestMismatch, hasMissingPkgs, errStatus
```

Comment:
With the named return vars, you can just do a bare `return` when the values are already set.

Reply:
yep, but I think it's better to do explicit returns when the function is big, for clarity.

Reply:
In that case, can we change most of these to return constant `true` or `false` for both vars? It looks like only the final `hasMissingPkgs` is actually dynamic.

Reply:
changed and pushed!